packages/nx-extensions/src/executors/assert-built-esm-and-cjs/executor.ts

```diff
@@ -0,0 +1,54 @@
+import { ExecutorContext } from '@nrwl/devkit';
+import { spawnSync } from 'child_process';
+import { mkdirSync, writeFileSync } from 'fs';
+import * as path from 'path';
+import { AssertBuiltEsmAndCjsExecutorSchema } from './schema';
+
+/**
+ * Test whether a module can be imported as both ESM and CJS.
+ *
+ * @param options
+ * @param context
+ * @returns
+ */
+export default async function runExecutor(
+	options: AssertBuiltEsmAndCjsExecutorSchema,
+	context: ExecutorContext
+) {
+	const buildDir = options.outputPath.split('/')[0];
+	const testsPath = path.join(context.root, buildDir, 'test-esm-cjs');
+	mkdirSync(testsPath, { recursive: true });
+
+	writeFileSync(
+		path.join(testsPath, 'test-esm.mjs'),
+		`import * as result from '../../${options.outputPath}/index.js';`
+	);
+	writeFileSync(
+		path.join(testsPath, 'test-cjs.cjs'),
+		`require('../../${options.outputPath}');`
+	);
+	writeFileSync(
+		path.join(testsPath, 'ensure-both-outputs-run-in-node.sh'),
+		`#!/bin/sh
+set -e
+node test-esm.mjs;
+node test-cjs.cjs;
+echo Success;`
+	);
+
+	const test = spawnSync('sh', ['ensure-both-outputs-run-in-node.sh'], {
+		cwd: testsPath,
+		stdio: 'pipe',
+		encoding: 'utf-8',
+	});
+	// Trim spaces and newlines
+	const stdout = test.output.toString();
+	const success =
+		test.status === 0 && stdout.replace(/[^A-Za-z]/g, '') === 'Success';
+	if (!success) {
+		throw `${context.targetName} could not be imported as both ESM and CJS: ${stdout}`;
+	}
+	return {
+		success,
+	};
```

Comment:
n case you find it useful, I believe we could also spawn `node` directly here _as the shell_/process, and then if we care we could run these in parallel via `spawn` instead of `spawnSync`

right now we're writing a shell wrapper to fail if the scripts fail, but then checking in node if the wrapper failed, when we could skip the middleman and check solely if the scripts themselves failed.

```js
const checkForSuccess = scriptName => new Promise( ( resolve, reject ) => {
	const test = spawn('node', [scriptName], { cwd: testsPath, stdio: 'pipe', encoding: 'utf-8' } );

	let stdout = '';
	let stderr = '';

	test.stdout.on('data', chunk => stdout += chunk);
	test.stderr.on('data', chunk => stderr += chunk);

	test.on('close', statusCode => {
		const didSucceed = statusCode === 0 && stdout.replace(/[^A-Z]/ig, '') === 'Success' );

		return didSucceed ? resolve(stdout) : reject(stderr);
	});
} );

const testRuns = await Promise.all( ['test-esm.mjs', 'test-cjs.cjs'].map( checkForSuccess ) );
```

Reply:
Nice! Done in e44bef00